addon/components/polaris-banner.js

```diff
@@ -96,109 +79,117 @@ export default class PolarisBanner extends Component {
-    }
-
-    return bannerIcons[status].color;
-  }
+  handleMouseUpByBlurring = handleMouseUpByBlurring;
 
-  @computed('title')
   get headingId() {
-    if (isBlank(this.title)) {
+    if (isBlank(this.args.title)) {
       return null;
     }
 
-    return `${guidFor(this)}-heading`;
+    return `${guidFor(this)}Heading`;
   }
 
-  @computed('status')
   get statusClass() {
-    let { status } = this;
+    let { status } = this.args;
     if (isBlank(status) || !supportedStatuses.includes(status)) {
       return null;
     }
 
     return `Polaris-Banner--status${capitalize(status)}`;
   }
 
-  @computed('statusClass', 'hasDismiss', 'withinContentContainer', 'class')
   get cssClasses() {
     let cssClasses = ['Polaris-Banner'];
     if (this.statusClass) {
       cssClasses.push(this.statusClass);
     }
-    if (this.hasDismiss) {
+    if (this.args.onDismiss) {
       cssClasses.push('Polaris-Banner--hasDismiss');
     }
+
+    if (this.shouldShowFocus) {
+      cssClasses.push('Polaris-Banner--keyFocused');
+    }
+
     if (this.withinContentContainer) {
       cssClasses.push('Polaris-Banner--withinContentContainer');
     } else {
       cssClasses.push('Polaris-Banner--withinPage');
     }
-    if (this.class) {
-      cssClasses.push(this.class);
+
+    if (this.polaris.features.newDesignLanguage) {
+      cssClasses.push('Polaris-Banner--newDesignLanguage');
     }
 
     return cssClasses.join(' ');
   }
 
-  init() {
-    super.init(...arguments);
-
-    deprecate(
-      `[PolarisBanner] Passing 'action' is deprecated! Please use 'primaryAction' instead`,
-      !this.action,
-      {
-        id: 'ember-polaris.polaris-banner.action-arg',
-        until: '7.0.0',
-      }
-    );
+  get useBannerAttributes() {
+    let { newDesignLanguage } = this.polaris.features;
+
+    switch (this.args.status) {
+      case 'success':
+        return {
+          defaultIcon: 'CircleTickMajor',
+          iconColor: newDesignLanguage ? 'success' : 'greenDark',
+          ariaRoleType: 'status',
+        };
+
+      case 'info':
+        return {
+          defaultIcon: 'CircleInformationMajor',
+          iconColor: newDesignLanguage ? 'highlight' : 'tealDark',
+          ariaRoleType: 'status',
+        };
+
+      case 'warning':
+        return {
+          defaultIcon: 'CircleAlertMajor',
+          iconColor: newDesignLanguage ? 'warning' : 'yellowDark',
+          ariaRoleType: 'alert',
+        };
+
+      case 'critical':
+        return {
+          defaultIcon: 'CircleDisabledMajor',
+          iconColor: newDesignLanguage ? 'critical' : 'redDark',
+          ariaRoleType: 'alert',
+        };
+
+      default:
+        return {
+          defaultIcon: newDesignLanguage
+            ? 'CircleInformationMajor'
+            : 'FlagMajor',
+          iconColor: newDesignLanguage ? 'base' : 'inkLighter',
+          ariaRoleType: 'status',
+        };
+    }
+  }
+
+  @action
+  handleBlur(e) {
+    e.currentTarget.blur();
```

Comment:
I don't see this in the react implementation, is this needed?

Reply:
Ah, given that this was missing from the `handleKeyUp`, I'm guessing I got my eyes crossed and put it in here instead of the mouseup method 🤦 Yeah I'll remove.